spectree/spec.py

```diff
@@ -335,12 +335,15 @@ def _generate_spec(self) -> Dict[str, Any]:
 
         if self.config.servers:
             spec["servers"] = [
-                server.dict(exclude_none=True) for server in self.config.servers
+                server.model_dump(exclude_none=True, mode="json")
```

Comment:
previously `.dict()` was returning a Python dictionary, now `.model_dump(mode="json")` returns json. Is this change intentional and if so, what difference does it make (I'm not familiar with this part of codebase, so I'd really like to know)?

Reply:
`mode="json"` just serializes the values for the returned dict to make sure there's no python objects so when it's jsonified for `openapi.json`, it won't cause any `_ is not JSONSerializable` errors to pop up when accessing apidocs. I noticed it happening with some openapi config models in general when running the falcon_demo.

in this case though and probably elsewhere, I might have overused it since `Server` model doesn't have any special types